File: fixtures/scale/java/large/src/cryptobench/ciphers/keywordsub/KeywordSubKey.java
package cryptobench.ciphers.keywordsub;

public class KeywordSubKey {
    private final String keyword;

    public KeywordSubKey(String keyword) {
        this.keyword = keyword.toUpperCase();
    }

    /** Keyword first (duplicates dropped), then the remaining letters in order. */
    public String mixedAlphabet() {
        StringBuilder sb = new StringBuilder();
        String source = keyword + "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
        for (char c : source.toCharArray()) {
            if (c >= 'A' && c <= 'Z' && sb.indexOf(String.valueOf(c)) < 0) {
                sb.append(c);
            }
        }
        return sb.toString();
    }

    public static KeywordSubKey defaultKey() {
        return new KeywordSubKey("OBSIDIAN");
    }
}

File: fixtures/scale/scala/large/src/cryptobench/ciphers/keywordsub/KeywordSubKey.scala
package cryptobench.ciphers.keywordsub

final case class KeywordSubKey(keyword: String) {

  /** Keyword first (duplicates dropped), then the remaining letters in order. */
  def mixedAlphabet: String = {
    val sb = new StringBuilder
    for (c <- keyword.toUpperCase + "ABCDEFGHIJKLMNOPQRSTUVWXYZ") {
      if (c >= 'A' && c <= 'Z' && sb.indexOf(c.toString) < 0) sb.append(c)
    }
    sb.toString
  }
}

object KeywordSubKey {
  def default(): KeywordSubKey = KeywordSubKey("OBSIDIAN")
}
